biz.aQute.launcher/src/aQute/launcher/Launcher.java

```diff
@@ -824,7 +837,10 @@ public void run() {
 		if (!workingdir.isDirectory())
 			throw new IllegalArgumentException("Cannot create a working dir: " + workingdir);
 
-		p.setProperty(Constants.FRAMEWORK_STORAGE, workingdir.getAbsolutePath());
+		if (System.getProperty(Constants.FRAMEWORK_STORAGE) == null)
+			p.setProperty(Constants.FRAMEWORK_STORAGE, workingdir.getAbsolutePath());
+		else
+			p.setProperty(Constants.FRAMEWORK_STORAGE, System.getProperty(Constants.FRAMEWORK_STORAGE));
```

Comment:
So you want to process the storageDir set and then ignore that when -Dorg.osgi.framework.storage? What about the other  org.osgi.framework.* setting on the command line? Like clean?